metaflow/plugins/aws/step_functions/step_functions.py

```diff
@@ -899,6 +936,59 @@ def _get_retries(self, node):
 
         return max_user_code_retries, max_user_code_retries + max_error_retries
 
+    def _get_command_s3_path(self, node_name, deployment_id=None):
+        """Generate S3 path for storing commands."""
+        # Use datastore root with commands suffix
+        base_path = self.flow_datastore.datastore_root.rstrip("/") + "/commands"
+
+        # Return relative path (without s3:// prefix) for datastore compatibility
+        if base_path.startswith("s3://"):
+            base_path = base_path[5:]  # Remove s3:// prefix
+
+        # Use deployment-specific ID instead of run ID
+        if deployment_id is None:
+            # Use the instance deployment_id if available, otherwise generate a new one
+            deployment_id = getattr(self, "deployment_id", None)
+            if deployment_id is None:
+                import time
+
+                deployment_id = f"deployment-{int(time.time())}"
+
+        return f"{base_path}/{self.flow.name}/{deployment_id}/{node_name}/command.sh"
```

Comment:
Just making a note here that there might be a slight security concern with now having the entire batch command being pulled from a very guessable S3 location, as depending on the organization the bucket might have more lax permissions for access compared to who/what can deploy to SFN

This should not be a problem as is, as long as we keep the default for compression at false, and call out the implications in the release docs.